src/test/java/ngrok/runner/config/BaseNgrokRunnerMockedConfiguration.java

```diff
@@ -47,6 +48,24 @@ public NgrokApiClient mockedNgrokApiClient() throws IOException {
 
         when(ngrokApiClient.fetchTunnels()).thenReturn(ngrokTunnelsList.getTunnels());
 
+        if (runningPort == 0) {
+            when(ngrokApiClient.isResponding())
+                    .thenReturn(false)
+                    .thenReturn(true);
+            when(ngrokApiClient.fetchTunnels(anyInt())).thenCallRealMethod();
+        } else {
+            when(ngrokApiClient.isResponding())
+                    .thenReturn(true);
+
+            NgrokTunnel tunnel = ngrokTunnelsList.getTunnels().get(0);
+            tunnel.getConfig().setAddr(new URL("http://localhost:" + runningPort));
+            when(ngrokApiClient.fetchTunnels(runningPort))
+                    .thenReturn(Collections.singletonList(tunnel));
+        }
+
+        when(ngrokApiClient.startTunnel(anyInt(), anyString(), anyString()))
```

Comment:
I didn't want to make too much complexity in mocks.
But to be honest I was a bit lazy too.
Do you have any values you'd like?

Reply:
I meant something like:
```java
when(ngrokApiClient.startTunnel(eq(8080), eq("http"), anyString()))
        .thenReturn(ngrokTunnelsList.getTunnels().get(0));
when(ngrokApiClient.startTunnel(eq(8080), eq("https"), anyString()))
        .thenReturn(...
```

but you can leave it like this, because I got work in progress done in NgrokApiClient in order to cover full ngrok api, just need to add some tests so anyways I'll need to update this one :)

Reply:
I will let you do better in the next step ;)